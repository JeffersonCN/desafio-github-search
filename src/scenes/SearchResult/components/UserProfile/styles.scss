.userProfile {
    font-family: 'Raleway';
    font-weight: 300;
    max-width: 300px;
    margin-bottom: 50px;

    .avatar {
        width: 280px;
        max-width: 280px;
        border-radius: 2px;
        box-shadow: 0 0 4px 0 rgba(0, 0, 0, 0.4);
        display: block;
    }

    .nameHeading {
        font-size: 35px;
        max-width: 280px;
        margin: 12px 0 0;
        line-height: 35px;
    }

    .loginSubheading {
        font-size: 20px;
        color: #5c5c5c;
        margin: 7px 0 28px;
    }
}