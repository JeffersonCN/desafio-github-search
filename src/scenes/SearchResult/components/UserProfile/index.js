import React, { Component } from 'react';
import TextWithLeftIcon from '../TextWithLeftIcon';
import { colors, boxShadow } from '../../styles';

import organizationIcon from '../../assets/icons/organization-icon.svg';
import locationIcon from '../../assets/icons/location-icon.svg';
import starIcon from '../../assets/icons/star-icon.svg';
import repositoriesIcon from '../../assets/icons/repositories-icon.svg';
import followersIcon from '../../assets/icons/followers-icon.svg';

const style = {
    container: {
        fontFamily: "Raleway",
        fontWeight: 300,
        padding: "5px"
    },
    avatar: {
        maxWidth: "280px",
        borderRadius: "2px",
        boxShadow: boxShadow
    },
    nameHeader: {
        fontSize: "35px",
        maxWidth: "280px",
        margin: "10px 0 0",
        lineHeight: "35px"
    },
    loginSubheading: {
        fontSize: "20px",
        color: colors.gray,
        margin: "10px 0 25px"
    }
}

const user = {
    "login": "JeffersonCN",
    "id": 8918576,
    "node_id": "MDQ6VXNlcjg5MTg1NzY=",
    "avatar_url": "https://avatars1.githubusercontent.com/u/8918576?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/JeffersonCN",
    "html_url": "https://github.com/JeffersonCN",
    "followers_url": "https://api.github.com/users/JeffersonCN/followers",
    "following_url": "https://api.github.com/users/JeffersonCN/following{/other_user}",
    "gists_url": "https://api.github.com/users/JeffersonCN/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/JeffersonCN/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/JeffersonCN/subscriptions",
    "organizations_url": "https://api.github.com/users/JeffersonCN/orgs",
    "repos_url": "https://api.github.com/users/JeffersonCN/repos",
    "events_url": "https://api.github.com/users/JeffersonCN/events{/privacy}",
    "received_events_url": "https://api.github.com/users/JeffersonCN/received_events",
    "type": "User",
    "site_admin": false,
    "name": "Jefferson Nascimento",
    "company": "FATEC",
    "blog": "http://jeffersoncruz.com",
    "location": "Praia Grande, SP - Brazil",
    "email": null,
    "hireable": true,
    "bio": "Full-time Coding Student passionated for technology.",
    "public_repos": 23,
    "public_gists": 5,
    "followers": 17,
    "following": 32,
    "created_at": "2014-09-25T17:36:40Z",
    "updated_at": "2018-07-26T22:37:07Z"
}

class UserProfile extends Component {
    render() {
        return (
            <div className="UserProfile" style={style.container} >
                {/* Avatar */}
                <img src={user.avatar_url} alt="avatar do usuÃ¡rio no Github" style={style.avatar} />
                {/* Name */}
                <p style={style.nameHeader}>{user.name}</p>
                {/* Login */}
                <p style={style.loginSubheading}>{user.login}</p>

                <TextWithLeftIcon icon={organizationIcon}   text={user.company} />
                <TextWithLeftIcon icon={locationIcon}       text={user.location} />
                <TextWithLeftIcon icon={starIcon}           text={10} />
                <TextWithLeftIcon icon={repositoriesIcon}    text={user.public_repos} />
                <TextWithLeftIcon icon={followersIcon}      text={user.followers} />
            </div>
        );
    }
}

export default UserProfile;