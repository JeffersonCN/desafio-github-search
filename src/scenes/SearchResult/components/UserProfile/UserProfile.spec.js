import React from 'react';
import { shallow } from 'enzyme';
import UserProfile from './';

describe('Testing UserProfile Component', () => {
    it('shoud render correctly', () => {
        const user = {
            avatar_url: "https://api.adorable.io/avatars/285/abott@adorable.png",
            name: "Jefferson Nascimento",
            login: "jeffersoncn",
            company: "Concrete Solutions",
            location: "Praia Grande, Brazil",
            public_repos: 30,
            followers: 10
        }

        const wrapper = shallow(
            <UserProfile user={user} stars={10}/>
        );

        expect(wrapper).toMatchSnapshot();
    });
});