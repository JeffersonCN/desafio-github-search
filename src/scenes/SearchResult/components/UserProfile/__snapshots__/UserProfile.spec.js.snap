// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UserProfile Component shoud render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserProfile
    stars={10}
    user={
        Object {
            "avatar_url": "https://api.adorable.io/avatars/285/abott@adorable.png",
            "company": "Concrete Solutions",
            "followers": 10,
            "location": "Praia Grande, Brazil",
            "login": "jeffersoncn",
            "name": "Jefferson Nascimento",
            "public_repos": 30,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="avatar do usu치rio no Github"
          className={undefined}
          src="https://api.adorable.io/avatars/285/abott@adorable.png"
/>,
        <p
          className={undefined}
>
          Jefferson Nascimento
</p>,
        <p
          className={undefined}
>
          jeffersoncn
</p>,
        <TextWithLeftIcon
          icon="organization-icon.svg"
          text="Concrete Solutions"
/>,
        <TextWithLeftIcon
          icon="location-icon.svg"
          text="Praia Grande, Brazil"
/>,
        <TextWithLeftIcon
          icon="star-icon.svg"
          text={10}
/>,
        <TextWithLeftIcon
          icon="repositories-icon.svg"
          text={30}
/>,
        <TextWithLeftIcon
          icon="followers-icon.svg"
          text={10}
/>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "avatar do usu치rio no Github",
          "className": undefined,
          "src": "https://api.adorable.io/avatars/285/abott@adorable.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Jefferson Nascimento",
          "className": undefined,
        },
        "ref": null,
        "rendered": "Jefferson Nascimento",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "jeffersoncn",
          "className": undefined,
        },
        "ref": null,
        "rendered": "jeffersoncn",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "icon": "organization-icon.svg",
          "text": "Concrete Solutions",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "icon": "location-icon.svg",
          "text": "Praia Grande, Brazil",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "icon": "star-icon.svg",
          "text": 10,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "icon": "repositories-icon.svg",
          "text": 30,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "icon": "followers-icon.svg",
          "text": 10,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="avatar do usu치rio no Github"
            className={undefined}
            src="https://api.adorable.io/avatars/285/abott@adorable.png"
/>,
          <p
            className={undefined}
>
            Jefferson Nascimento
</p>,
          <p
            className={undefined}
>
            jeffersoncn
</p>,
          <TextWithLeftIcon
            icon="organization-icon.svg"
            text="Concrete Solutions"
/>,
          <TextWithLeftIcon
            icon="location-icon.svg"
            text="Praia Grande, Brazil"
/>,
          <TextWithLeftIcon
            icon="star-icon.svg"
            text={10}
/>,
          <TextWithLeftIcon
            icon="repositories-icon.svg"
            text={30}
/>,
          <TextWithLeftIcon
            icon="followers-icon.svg"
            text={10}
/>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "avatar do usu치rio no Github",
            "className": undefined,
            "src": "https://api.adorable.io/avatars/285/abott@adorable.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Jefferson Nascimento",
            "className": undefined,
          },
          "ref": null,
          "rendered": "Jefferson Nascimento",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "jeffersoncn",
            "className": undefined,
          },
          "ref": null,
          "rendered": "jeffersoncn",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "icon": "organization-icon.svg",
            "text": "Concrete Solutions",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "icon": "location-icon.svg",
            "text": "Praia Grande, Brazil",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "icon": "star-icon.svg",
            "text": 10,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "icon": "repositories-icon.svg",
            "text": 30,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "icon": "followers-icon.svg",
            "text": 10,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
